src/api/spec/models/bs_request_spec.rb

```diff
@@ -156,6 +156,42 @@
         expect(request.history_elements.first.type).to eq('HistoryElement::RequestDeleted')
       end
     end
+
+    context 'final state declined cannot be changed ' do
+      let(:request) do
+        create(:declined_bs_request,
+               target_package: target_package,
+               target_project: target_project,
+               source_package: source_package,
+               source_project: source_project)
+      end
+      let(:user) { create(:confirmed_user) }
+
+      before do
+        User.current = user
+      end
+
+      it { expect { request.change_state(newstate: 'review') }.to raise_error(PostRequestNoPermission) }
+    end
+
+    context 'final state accepted cannot be changed' do
+      let!(:request) do
+        create(:bs_request_with_submit_action,
+               target_project: target_project.name,
+               source_project: source_package.project.name,
+               source_package: source_package.name)
+      end
+      let!(:relationship_project_user) { create(:relationship_project_user, project: target_project) }
+      let(:user) { relationship_project_user.user }
+      before do
+        User.current = user
+        request.state = 'accepted'
+        request.save
+      end
+
+      it { expect(request.state).to eq(:accepted) }
+      it { expect { request.change_state(newstate: 'review') }.to raise_error(PostRequestNoPermission) }
```

Comment:
How do you know that this is not the same cause as in the test above or any other thing that raises `PostRequestNoPermission,`?

You are also missing the test to verify that `request.state` is still in accepted _after_ you called `.change_state`.

Reply:
actually because the randomization of the "it" blocks, I was testing it 50% of the time after :)